### PR TITLE
typings: refactor to support unified

### PR DESCRIPTION
**Pull request by ChristianMurphy**
also:
* removes nullish
* adds comments
* adds more type tests
* clarifies which props can be used with reset

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,46 +1,66 @@
-// TypeScript Version: 3.7
+// TypeScript Version: 3.3
```
Comment (ChristianMurphy) [inline index.d.ts:1]: I set this down to the lowest version that passed dtslint

File: index.d.ts [hunk 2]
```diff
@@ -65,32 +90,26 @@ declare namespace messageControl {
...
-     * whether to treat all messages as turned off initially
-     * defeaults to false.
-     */
-    reset?: boolean | Nullish
-    /**
-     * list of `ruleId`s to initially turn on if `reset: true`.
-     */
-    enable?: string[] | Nullish | unknown
-    /** list of `ruleId`s to turn off if `reset: false` */
-    disable?: string[] | Nullish | unknown
```
Comment (ChristianMurphy) [inline index.d.ts:79]: These were moved to the sub-classes, to allow for more fine grained typings

File: type-test.ts [hunk 3]
```diff
@@ -3,12 +3,127 @@
...
+  disable: ['rule-id']
+})
+
+control({
+  name: 'foo',
+  marker: (n: Node) => null,
+  test: 'html',
+  reset: false,
+  // $ExpectError
+  enable: ['rule-id']
```
Comment (ChristianMurphy) [inline type-test.ts:53]: added tests for variations of props

File: type-test.ts [hunk 4]
```diff
@@ -3,12 +3,127 @@
...
+control({
+  name: 'foo',
+  marker: (n: Node) => null,
+  test: 'html',
+  reset: true,
+  // $ExpectError
+  disable: ['rule-id']
+})
+
+unified().use(control, {
```
Comment (ChristianMurphy) [inline type-test.ts:76]: added tests for usage with unified

File: index.d.ts (hunk 2 above)
Comment (Zemnmez) [reply index.d.ts:79]: honestly thinking about it this is more correct than the approaches I was trying

File: index.d.ts (hunk 1 above)
Comment (Zemnmez) [reply index.d.ts:1]: what's the command for this! this would be super useful :)

File: type-test.ts (hunk 4 above)
Comment (Zemnmez) [reply type-test.ts:76]: nice! thanks :)

**Review by Zemnmez — APPROVED** [review]

**Pull request merged**